src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -144,8 +146,43 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 			this._layoutCalledOnce = true;
 		}
 	}
+	reorderGridData(source: IDataResource) {
+		let rowIndex = 0;
+		// Get schema list of grid
+		const columnNames: string[] = source.schema.fields.map(field => field.name);
+		if (source.data.length > 0) {
+			// Check to see if data source is ordered properly based on schema
+			let rowKeys = Object.keys(source.data[0]);
+			if (!equals(columnNames, rowKeys)) {
+				// SQL notebooks does not use columnName as key (instead uses indices)
+				// Indicies indicate the row is ordered properly
+				// We check the data to know if it is in index form or string form
+				for (let index of rowKeys) {
+					// Check to see if first row keys are in index format (numbers)
+					let currentIndex = typeof Number(index);
+					let nextIndex = typeof Number(rowKeys[currentIndex + 1]);
+					if (currentIndex === 'number' && rowKeys[nextIndex] < (rowKeys.length - 1)) {
+						//check to see if next element is number (confirm table is already ordered)
+						if (!(nextIndex === 'number')) {
+							this._notificationService.error(localize('orderFailed', "Not able to have mix of numbers and string column names"));
```

Comment:
It seems weird to have this check here instead of doing this all at the start and just exiting out early if they're all numbers.

Reply:
It also seems weird to have this error and then not really doing anything about it. Either error out completely or skip re-ordering I would think

Reply:
I think no matter what this has to stay inside the if condition `!equals(columnNames,rowKeys)` as SQL and other papermill executed grids (currently) will trigger this. We should be checking each row current and its nextIndex to make sure it is in index format. 

I think skipping re-ordering may be best.